lib/sedna/core/lifelong_learning/lifelong_learning.py

```diff
@@ -165,95 +224,113 @@ def train(self, train_data,
-                                                              save_model)
-            model_file = model_upload_key[model_file]
-
-            try:
-                model = self.kb_server.upload_file(save_model)
-            except Exception as err:
-                self.log.error(
-                    f"Upload task model of {model_file} fail: {err}"
-                )
-                model = set_backend(
-                    estimator=self.estimator.estimator.base_model
-                )
-                model.load(model_file)
-            task.model.model = model
-
-            for _task in task.tasks:
-                sample_dir = FileOps.join_path(
-                    self.config.output_url,
-                    f"{_task.samples.data_type}_{_task.entry}.sample")
-                task.samples.save(sample_dir)
-                try:
-                    sample_dir = self.kb_server.upload_file(sample_dir)
-                except Exception as err:
-                    self.log.error(
-                        f"Upload task samples of {_task.entry} fail: {err}")
-                _task.samples.data_url = sample_dir
-
-        save_extractor = FileOps.join_path(
-            self.config.output_url,
-            KBResourceConstant.TASK_EXTRACTOR_NAME.value
-        )
-        extractor = FileOps.dump(extractor, save_extractor)
-        try:
-            extractor = self.kb_server.upload_file(extractor)
-        except Exception as err:
-            self.log.error(f"Upload task extractor fail: {err}")
-        task_info = {
-            "task_groups": task_groups,
-            "extractor": extractor
-        }
-        fd, name = tempfile.mkstemp()
-        FileOps.dump(task_info, name)
+        task_index = dict(
+            seen_task=seen_task_index,
+            unseen_task=unseen_task_index)
+        task_index_url = FileOps.dump(
+            task_index, self.cloud_knowledge_management.local_task_index_url)
 
-        index_file = self.kb_server.update_db(name)
-        if not index_file:
-            self.log.error(f"KB update Fail !")
-            index_file = name
-        FileOps.upload(index_file, self.config.task_index)
+        task_index = self.cloud_knowledge_management.update_kb(
+            task_index_url, self.kb_server)
+        res.update(unseen_res)
 
         task_info_res = self.estimator.model_info(
-            self.config.task_index,
+            task_index,
             relpath=self.config.data_path_prefix)
         self.report_task_info(
             None, K8sResourceKindStatus.COMPLETED.value, task_info_res)
         self.log.info(f"Lifelong learning Train task Finished, "
-                      f"KB idnex save in {self.config.task_index}")
+                      f"KB index save in {task_index}")
         return callback_func(self.estimator, res) if callback_func else res
 
     def update(self, train_data, valid_data=None, post_process=None, **kwargs):
```

Comment:
combine this func`update` and `train` into an external func, e,g,: 
```python
def train(self):
    if not has_completed_initial_training:
        return self._initial_train()
    return self._update(self)
      
``` 
Combine A and B into an external function.